File: daily_action_cycle_mobile_ui/lib/core/entities/activity.dart
import 'package:equatable/equatable.dart';

class Activity extends Equatable {
  final String id;
  final String title;
  final String description;
  final DateTime createdAt;
  final DateTime dueDate;
  final bool isCompleted;
  final bool isNotified;
  final DateTime? updatedAt;
  final bool isDeleted;
  final DateTime? deletedAt;

  Activity({
    required this.id,
    required this.title,
    required this.description,
    required this.createdAt,
    required this.dueDate,
    this.isCompleted = false,
    this.isNotified = false,
    this.updatedAt,
    this.isDeleted = false,
    this.deletedAt,
  });

  @override
  List<Object?> get props => [
        id,
        title,
        description,
        createdAt,
        dueDate,
        isCompleted,
        isNotified,
        updatedAt,
        isDeleted,
        deletedAt
      ];
}

File: DailyActionCycle.WebAPI/DailyActionCycle.WebUI/Entities/Activity.cs
namespace DailyActionCycle.WebUI.Entities;

public sealed record class Activity
{
    public Guid Id { get; init; } = Guid.NewGuid();

    public required string Title { get; set; }

    public required string Description { get; set; }

    public DateTime CreatedAt { get; init; } = DateTime.UtcNow;

    public DateTime DueDate { get; set; } = DateTime.UtcNow.AddHours(1);

    public bool IsCompleted { get; set; } = false;

    public bool IsNotified { get; set; } = true;


    public DateTime? UpdatedAt { get; set; }

    public bool IsDeleted { get; set; } = false;

    public DateTime? DeletedAt { get; set; }


    public void Complete() => IsCompleted = true;
}
